src/js/components/AppListComponent.jsx

```diff
@@ -80,12 +113,42 @@ var AppListComponent = React.createClass({
     });
   },
 
+  getAppsInGroup: function () {
+    var apps = this.state.apps;
+    var currentGroup = this.props.currentGroup;
+
+    return lazy(apps)
+      .filter((app) => app.id.startsWith(currentGroup))
+      .reduce((memo, app) => {
```

Comment:
Cool. :)

Note on 3., putting the groups on top is also just a sorting to me. So this should be handled where the other sorting of the columns also takes places. Should be easy because of this isGroup-property.